### DISCUSSION TITLE
Attributter - Virksomhet - legal_entity_id/legal_entity_name og point_of_care_id/point_of_care_name

### DISCUSSION THREAD

**Discussion by eirikbroen** [NorskHelsenett/Tillitsrammeverk#12] [General]
Antar at disse attributtene tar over for _organization_ og _child-organization_. Navnet spiller ikke så stor rolle her og flott med bevegelse i retning av internasjonal standard. Det jeg lurer på er om vi burde gjøre _point_of_care_id_ obligatorisk? Ser kilde til _legal_entity_id_ er HelseID/Altinn, men er det slik at denne verdien i praksis vil avledes fra org.nr for virksomhetene som eier konto i Selvbetjeningsløsningen hvor HelseID klient er opprettet? 

Er det slik at vi burde endre kilde for  _point_of_care_id_ og _point_of_care_name_ til å være HelseID klient (ofte EPJ eller en mellomvare hos aktør)? Kan være nyttig å kunne forutsette at det alltid finnes en definert verdi for _point_of_care_ selv om den i noen tilfeller vil være samme verdi som _legal_entity_

**Comment by steinarnoem** [comment]
Vi har lagt til en tabell som uttrykker informasjonskildene litt tydeligere i spesifikasjonen nå: https://github.com/NorskHelsenett/Tillitsrammeverk/blob/main/specs/informasjons_og_datamodell.md#323-oppsummering-av-p%C3%A5krevd-eller-valgfri-informasjon

HelseID tilbyr et løsningsmønster som er tilpasset §9 samarbeid og multi-tenancy løsninger, hvor fellesnevneren er at flere dataansvarlige bruker samme system, som driftes av en databehandler. Dette løsningsmønsteret forutsetter at den dataansvarlige har utført en eksplisitt delegering hvor deres databehandler har lov til å opptre på vegne av seg. Delegeringen utføres i dag i Altinn.

For slike systemer og samarbeid må systemet, ved hver forespørsel om tilgang i HelseID, angi den dataansvarlige som helsepersonellet yter helsehjelp hos. HelseID kontrollerer at det foreligger en delegering for bruk av HelseID - altså at den angitte dataansvarlige har delegert denne retten til sin databehandler i Altinn.

Systemet kan også angi behandlingsstedet (i kommunal sammenheng kan dette være et sykehjem eller legevakt osv) i forespørselen.
Jeg ser for meg at denne mekanismen gjenbrukes for dokumentdeling, og at HelseID reflekterer dataansvarlig og behandlingssted i henhold til datamodellen i forespørselen og i tokens som blir utstedt.

@eirikbroen, kan du ta en titt på datamodelldokumentet etter endringene? Ser det bedre ut nå?

**Reply by stvgr** [reply]
Dette er OK for Stavanger, vi tenker at det er viktig at innbygger i sin logg ser behandlingsstedet.

**Reply by NEIsaksen** [reply]
Vet ikke om det angår andre store byer med bydeler enn Oslo, men for oss i CGM så ser det ut til at i Oslo så er det Bydelene som er parent Org. og child org er f.eks Helsestasjonen bydel Sagene.
Det samme gjelder slik vi kan se for kommunale legekontor. Altså at bydelen er parentorg og legekontoret sitt orgno er child.
Dette mistenker vi blir ett mareritt å sette opp. Da man gjerne har samme system for kommunen mtp Helsestasjonstjenestene og antageligvis også PLO. 
I tillegg har man jo andre tjenester enn dette, f.eks fengsler, migrasjonstjenster, smitteverntjenester osv osv.

I EDI så er det stort sett Kommunen sitt OrgNo som benyttes for nivå 1 (om det ikke er interkommunal) adressering, og som kunder pr idag har lagt inn i systemet. Det er ikke idag vanlig for våre kunder å ha OrgNo annet enn på kommune nivå. 
Dette må kundene nok starte å legge dette inn for underenheter også fremover mtp HelseId og nå også Tillitsanker.

Noen andre som ser det samme issuet som oss i CGM her med OrgNo problemet mtp HelseId og da også Tillitsanker her?

**Reply by eirikbroen** [reply]
@NEIsaksen litt usikker på hvilket oppsett du ser på, men min antakelse er vel at Oslo ønsker å ha org.nr for Oslo Kommune eller Helseetaten som legal entity (parent) og org.nr for den enkelte tjeneste, f.eks en helsestasjon som point of care (child). 

Verdi for parent kommer direkte fra HelseID konfigurasjon og verdi for child fra EPJ.

**Reply by NEIsaksen** [reply]
@eirikbroen 
brreg -> https://w2.brreg.no/enhet/sok/underenh.jsp?orgnr=997506499
Helseetaten ser ut til å ha Legevakter og KAD.
Helsestasjon og Pleie og Omsorg ligger pr Bydel. 
Det samme med Kommunale legekontor ala Sagene Lokalmedisinske Senter.
Se her for f.eks Bydel Sagene -> https://w2.brreg.no/enhet/sok/underenh.jsp?orgnr=974778726

**Reply by NEIsaksen** [reply]
Ønsker å ta ett par eksempler her.

1: Helsestasjonssystemet i Oslo Kommune:

OrgNo registrert hos dem og i NHN AR er -> OSLO KOMMUNE HELSEETATEN 997506499.
Dette er eneste OrgNo registrert i EPJ! 

De har jo en enhetsstruktur hvor de rollemessig kan logge seg inn på Helsestasjon Bydel Sagene.
Når vi nå innfører HelseId og etterhvert Tillitsanker så blir dette litt rart.
Bydel Sagene ligger ikke under Oslo Kommune Helseetaten.
Helsestasjonen hvor helsepersonell faktisk logger seg inn vil da ligge på Helsestasjon Bydel Sagene som igjen ligger under Bydel Sagene.
Når epj ev. skal avlevere denne infoen gjennom tillitsankeret så blir jo dette meget rart.

All EDI går med OrgNo Helseetaten + HerId på helseetaten som nivå 1, og tjenestens HERID altså f.eks Helsestasjonstjenesten som nivå 2.
Så i utgangspunktet så er det denne infoen som eksisterer hos dem.

2: Kommunalt fastlegekontor- Sagene Lokalmedisinske Senter:

De er registrert i EPJ med OrgNo. 995 415 984.
Dette er eneste OrgNo registrert i EPJ pt! 
Deres "Parent" Organisasjon er "OSLO KOMMUNE BYDEL 3 SAGENE" med OrgNo 974 778 726.

All EDI går med OrgNo Sagene Lokalmedisinske Senter+ HerId på nivå 1, og fastlegene sin HERID nivå 2.

Så i utgangspunktet så er det denne infoen som eksisterer i EPJ hos dem.

---

Poenget er at dette med å benytte OrgNo (parent<-> child) i tillitsanker og kanskje spesielt HelseId er ikke alltid like trivielt.

**Reply by eirikbroen** [reply]
Hei @NEIsaksen, takk for eksempel. Du har helt rett i at KAD og Legevakten organisatorisk ligger under Helseetaten mens for eksempel hjemmehjelpstjenester ligger under bydelene. Det er for tiden en diskusjon rundt hvordan Oslo skal representere sine tjenester når vi skal benytte dette nye Tillitsrammeverket og jeg kan dessverre ikke gi deg konklusjonen helt ennå. Det jeg kan si er at en mulig løsninger her er at HelseID klient både for applikasjoner i bruk i tjenester under Helseetaten, Sykehjemsetaten eller bydel registreres på Oslo Kommune.

Det vil da innebære at for kontekstinformasjon utvekslet via HelseID som tillitsanker vil verdi for _legal_entity_ være:

```
"legal_entity": {
	"id": "958935420",
	"name": "OSLO KOMMUNE",
	"system": "urn:oid:2.16.578.1.12.4.1.4.101",
	"authority": "www.brreg.no"
} 
``` 

og for Sagene Lokalmedisinske Senter vil verdi for _point_of_care_ være:

```
"point_of_care": {
	"id": "995415984",
	"name": "SAGENE LOKALMEDISINSKE SENTER",
	"system": "2.16.578.1.12.4.1.4.101",
	"authority": "www.brreg.no"
}
```

Ved å angi Oslo kommune som øverste juridiske enhet er det mulig å traversere organisasjonsstruktur via API fra brreg.
Fra 958935420 - OSLO KOMMUNE

[https://data.brreg.no/enhetsregisteret/api/enheter?overordnetEnhet=958935420](https://data.brreg.no/enhetsregisteret/api/enheter?overordnetEnhet=958935420)

via 876819872 - OSLO KOMMUNE, BYRÅDSAVDELING FOR HELSE, ELDRE OG INNBYGGERTJENESTER

[https://data.brreg.no/enhetsregisteret/api/enheter?overordnetEnhet=876819872](https://data.brreg.no/enhetsregisteret/api/enheter?overordnetEnhet=876819872)

og til slutt for å finne Sagene Lokalmedisinske som underenhet av 974778726 - OSLO KOMMUNE BYDEL 3 SAGENE

[https://data.brreg.no/enhetsregisteret/api/underenheter?overordnetEnhet=974778726&size=100](https://data.brreg.no/enhetsregisteret/api/underenheter?overordnetEnhet=974778726&size=100)

Bruk av HelseID og tillitsrammeverket vil følgelig skille seg noe fra EDI og noe av konfigurasjon vil som nevnt over gjøres ved opprettelse av HelseID klient og noe vil konfigureres i EPJ. 

Fortsetter gjerne denne diskusjonen slik at vi kan legge til rette for en effektiv implementasjon av Tillitsrammeverket i EPJ.
